packages/backend/src/plugins/app.ts

```diff
@@ -1,14 +1,65 @@
 import { createRouter } from '@backstage/plugin-app-backend';
-import type { Router } from 'express';
+import { Router } from 'express';
 import type { PluginEnvironment } from '../types';
+import { resolvePackagePath } from '@backstage/backend-common';
+import { resolve as resolvePath } from 'path';
+import fs from 'fs-extra';
+import rateLimit from 'express-rate-limit';
 
 export default async function createPlugin(
   env: PluginEnvironment,
 ): Promise<Router> {
-  return await createRouter({
+  const appPackageName = 'app';
+
+  const appDistDir = resolvePackagePath(appPackageName, 'dist');
+  const staticDir = resolvePath(appDistDir, 'static');
+
+  const files = await fs.readdir(staticDir);
+  const jsFiles = files.filter(file => file.endsWith('.js'));
+  let injectedJSFile: string | undefined = undefined;
+
+  for (const jsFile of jsFiles) {
+    const path = resolvePath(staticDir, jsFile);
+
+    const content = await fs.readFile(path, 'utf8');
+    if (content.includes('__APP_INJECTED_')) {
+      injectedJSFile = jsFile;
+      break;
```

Comment:
Is it 100% confirmed that there will only ever be one file with the mark? I don't think its that much extra work to assume there can be more than one file that gest mutated at runtime.

Reply:
> Is it 100% confirmed that there will only ever be one file with the mark

It is how the `backend-app` plugins works. See the `return` lines in https://github.com/backstage/backstage/blob/master/plugins/app-backend/src/lib/config.ts#L55 and https://github.com/backstage/backstage/blob/master/plugins/app-backend/src/lib/config.ts#L64

Reply:
OK. Well. Then we are good.